src/App.js

```diff
@@ -1,86 +1,183 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
+import { InputField } from './components/InputField';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { FILTERS } from './constants';
+
+const useLocalStorage = (key, initialValue) => {
+  const [value, setValue] = useState(() => {
+    try {
+      return JSON.parse(localStorage.getItem(key)) || initialValue;
+    } catch {
+      return initialValue;
+    }
+  });
+
+  const saveValue = (newValue) => {
+    setValue(newValue);
+    localStorage.setItem(key, JSON.stringify(newValue));
+  };
+
+  return [value, saveValue];
+};
 
 function App() {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+  const [allStatus, setAllStatus] = useState(false);
+  const [filter, setFilter] = useState(FILTERS.all);
+
+  const getTodos = (todosList) => {
+    switch (filter) {
+      case (FILTERS.active):
+        return todosList.filter(todo => todo.completed === false);
+
+      case (FILTERS.completed):
+        return todosList.filter(todo => todo.completed === true);
+
+      default:
+        return todosList;
+    }
+  };
+
+  const filteredTodos = useMemo(() => getTodos(todos), [todos, filter]);
+
+  const addNewTodo = (newTodo) => {
+    setTodos([newTodo, ...todos]);
```

Comment:
use callback in setTodos
```suggestion
    setTodos(prevTodos => [newTodo, ...prevTodos]);
```